### ISSUE TITLE
NSVisualEffectView/Vibrancy in GPUI & Zed

### ISSUE THREAD

**Issue by nshout** [zed-industries/zed#7955]
### Check for existing issues

- [X] Completed

### Describe the feature

This feature request entails enhancing the visual aesthetics of the Zed application, particularly on macOS, by incorporating modern design elements such as transparent/blurred sidebars and a transparent/blurry title bar similar to Safari's compact mode. This would involve integrating NSVisualEffectView, a native macOS feature, into GPUI.

Expanding on GPUI as a Metal rendering-first GUI framework, the proposed feature should initially be exclusive to macOS to streamline development as other platforms have different ways to achieve a similar transparent/blurry design. Windows has Acrylic and Mica, Wayland and Xorg have god knows what, perhaps the DE that controls it.

TL:DR
Implement transparent/blurred sidebars and a transparent/blurry title bar option in Zed and GPUI as an overall option on div's, inspired by macOS design aesthetics, using NSVisualEffectView integration within GPUI.

[NSVisualEffectView Documentation](https://developer.apple.com/documentation/appkit/nsvisualeffectview)

### If applicable, add mockups / screenshots to help present your vision of the feature

<img width="594" alt="NSVisual" src="https://github.com/zed-industries/zed/assets/61021151/b6210c04-706f-4567-9c53-15a2f6e3b14b">

**Comment by jansol** [comment]
This is being tracked in #5040

**Comment by JosephTLyons** [comment]
Hey @nshout, let's close this out in favor of #5040. If #5040 doesn't quite cover all the things you are wanting, feel free to open a new issue with the difference.
